client.lua

```diff
@@ -751,18 +769,36 @@ end)
 CreateThread(function()
     while true do
         Citizen.Wait(2000)
-        local playerCoords = GetEntityCoords(PlayerPedId())
-        local distanceFromPed = #(POSTAL_BOSS_COORDS - playerCoords)
-
-        if distanceFromPed < 200 and not isPedSpawned then
-            isPedSpawned = true
-            spawnPostalBossPed()
-        end
+		if (Config.FRAMEWORK == 'esx') then
```

Comment:
this change would break `qb` compatibility. Here's what I propose to decouple and for us to be more `pure`! 

modify this thread to be
```lua
CreateThread(function()
    local spawnFunction = Config.FRAMEWORK == 'qb' and handlePedSpawnQB or handlePedSpawnESX
    while true do
        Wait(2000)
        spawnFunction()
    end
end)
```

we make two functions
```lua
function handlePedSpawnESX()
    local isLoaded = FRAMEWORK.IsPlayerLoaded()
    
    if not isLoaded then
        return
    end
    
    local playerCoords = GetEntityCoords(PlayerPedId())
    local distanceFromPed = #(POSTAL_BOSS_COORDS - playerCoords)

    if distanceFromPed < 200 and not isPedSpawned then
        isPedSpawned = true
        spawnPostalBossPed()
    end

    if postalBossPed and distanceFromPed >= 200 and isPedSpawned then
        isPedSpawned = false
        DeletePed(postalBossPed)
    end
end
```

```lua
function handlePedSpawnQB()
        local playerCoords = GetEntityCoords(PlayerPedId())
        local distanceFromPed = #(POSTAL_BOSS_COORDS - playerCoords)

        if distanceFromPed < 200 and not isPedSpawned then
            isPedSpawned = true
            spawnPostalBossPed()
        end

        if postalBossPed and distanceFromPed >= 200 and isPedSpawned then
            isPedSpawned = false
            DeletePed(postalBossPed)
        end
end
```

Let me know your thoughts!
@Misha0717 If you see this notification would love for you to chime in and throw a second opinion :)

Reply:
Looks good imo. But I should change the `isPedSpawned` variable to `DoesEntityExist(postalBossPed)` instead of a boolean. The functions should show like this. 

```
function handlePedSpawnQB()
        local playerCoords = GetEntityCoords(PlayerPedId())
        local distanceFromPed = #(POSTAL_BOSS_COORDS - playerCoords)
        local isPedSpawned = DoesEntityExist(postalBossPed)

        if distanceFromPed < 200 and not isPedSpawned then
            spawnPostalBossPed()
        end

        if distanceFromPed >= 200 and isPedSpawned then
            DeletePed(postalBossPed)
        end
end
```

I have never worked with multicharacter from ESX so I don't know if the fix of @lush0n3 so fix the issue. But is he tested it and it works fine I think this pr could be merged